File: Gestion_Angular_Front_end/src/app/classes/produit-dto.ts

import { Component, OnInit } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { CategorieDto } from './categorie-dto';
import { MagasinDto } from './magasin-dto';
import { LigneFactureDto } from './lignefacture-dto';

export class ProduitDto {


  /** Identifiant unique du produit */
  id: number = 0;

  /** Code produit unique */
  codeproduit: string = "";

  /** Nom du produit */
  nom: string = "";

  /** Description détaillée du produit */
  designation: string = "";

  prixAchat: number = 0;

   categorieDto:CategorieDto= new CategorieDto();
//  ligneFactureDto:LigneFactureDto= new LigneFactureDto();


}

File: backend/src/main/java/com/odji/spring_back_end/dto/ProduitDto.java
package com.odji.spring_back_end.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ProduitDto {

    private Integer id;
    private String codeproduit;
    private String nom;
    private String designation;
    private BigDecimal quantite;

    // Relations sortantes exposées (utile côté front)
    private CategorieDto categorie;
    private MagasinDto magasin;

    // ⚠️ On N'EXPOSE PAS les relations inverses dans le DTO de Produit.
    //    Voir explication ci-dessous.
}
